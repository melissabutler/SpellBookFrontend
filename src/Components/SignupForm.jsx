import React, { useContext, useEffect } from "react";
import useFields from "../Hooks/useFields";
import { useNavigate } from "react-router-dom"
import CurrentUserContext from "../currentUserContext";

// import "./SignupForm.css"

const SignupForm = ({signUp}) => {
    const currentUser = useContext(CurrentUserContext)
    
    const navigate = useNavigate()

    const [formData, handleChange, resetFormData] = useFields({
        username: "",
        password: "",
        email: ""
    })

    const handleSubmit = e => {
        e.preventDefault();
        if(!formData.username || !formData.password || !formData.email){
            alert("Please fill out the form.")
        } else {
            let newUser = {
                "username": formData.username.toString(),
                "password": formData.password.toString(),
                "email": formData.email.toString()
            }
                signUp({newUser})
            
        }   
    }

    useEffect(function navIfSignUp() {
        async function checkUser() {
            if(currentUser){
                navigate('/')
            }
        }
        checkUser();
    }, [currentUser])

    return (
        <div className="SignupForm">
            <h1>Sign Up</h1>
            <form className="SignupForm-form" onSubmit={handleSubmit}>
                <div className="SignupForm-section">
                    <div className="SignupForm-section-label">
                        <label className="SignupForm-label" htmlFor="username">Username</label>
                    </div>
                    <input 
                        className="SignupForm-input"
                        id="username"
                        name="username"
                        type="text"
                        value={formData.username}
                        onChange={handleChange}
                        placeholder="Username"
                    />
                </div>
                
                <div className="SignupForm-section">
                    <div className="SignupForm-section-label">
                        <label className="SignupForm-label" htmlFor="password">Password</label>
                    </div>
                    
                    <input 
                        className="SignupForm-input"
                        id="password"
                        name="password"
                        type="text"
                        value={formData.password}
                        onChange={handleChange}
                        placeholder="Password"
                    />
                </div>
                
                <div className="SignupForm-section">
                        <div className="SignupForm-section-label">
                        <label className="SignupForm-label" htmlFor="username">Email</label>
                        </div>
                    
                    <input 
                        className="SignupForm-input"
                        id="email"
                        name="email"
                        type="email"
                        value={formData.email}
                        onChange={handleChange}
                        placeholder="Email"
                    />
                </div>
                

                <button>Sign Up</button>
            </form>
        </div>
    )
}

export default SignupForm;