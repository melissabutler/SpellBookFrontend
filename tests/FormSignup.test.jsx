import React from "react";
import { describe, it, expect, vi } from "vitest";
import { render, screen} from '@testing-library/react'
import SignupForm from "../src/Components/FormSignup";

describe("SignupForm", function() {

        vi.mock('react', async (importOriginal) => {
            const react = await importOriginal('react');
            return {
                ...react,
                useContext: () => ({ currentUser: null}),
                useRef: () => { target: null }
        }})

        vi.mock('react-router-dom', async () => {
            const mod = await vi.importActual('react-router-dom');
            const navigate = vi.fn()
            return {
                ...mod, 
                useNavigate: () => navigate,
                BrowserRouter: () => "",
            }
        })
    it('renders the LoginForm component', async () => {
        render(
            <SignupForm />)
        screen.debug();
      
    });

        it('matches snapshot', () => {
            const { asFragment } = render(<SignupForm />);
            expect(asFragment()).toMatchSnapshot();
        });



})
   