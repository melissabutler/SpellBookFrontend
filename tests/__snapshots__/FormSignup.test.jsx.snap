// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SignupForm > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Form container"
  >
    <div
      class="row"
    >
      <h1>
        Sign Up
      </h1>
    </div>
    <form
      class="Form-form"
    >
      <div
        class="Form-section-label row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <label
            class="Form-label"
            for="username"
          >
            Username
          </label>
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <input
            class="Form-input"
            id="username"
            name="username"
            placeholder="Username"
            type="text"
            value=""
          />
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="Form-section-label row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <label
            class="Form-label"
            for="password"
          >
            Password
          </label>
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <input
            class="Form-input"
            id="password"
            name="password"
            placeholder="Password"
            type="text"
            value=""
          />
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="Form-section-label row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <label
            class="Form-label"
            for="username"
          >
            Email
          </label>
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <input
            class="SignupForm-input"
            id="email"
            name="email"
            placeholder="Email"
            type="email"
            value=""
          />
        </div>
        <div
          class="col"
        />
      </div>
      <div
        class="Form-section row"
      >
        <div
          class="col"
        />
        <div
          class="col"
        >
          <button>
            Sign Up
          </button>
        </div>
        <div
          class="col"
        />
      </div>
    </form>
  </div>
</DocumentFragment>
`;
